### DISCUSSION TITLE
preprocessing files during import - do I have to write a plugin?

### DISCUSSION THREAD

**Discussion by ctracy** [beetbox/beets#4475] [Q&A] [answered]
Is there any easy way to run simple shell commands against files when they are being imported?  Or do I need to write my own plugin?

I am relatively new to beets.  This came up because I was wondering if there was a simple way to run 'xattr -c' against every file being touched/etc in the directory that is being processed, instead of me having to remember to do it... (during import, my files are getting placed into an ExFAT filesystem, and I just kinda hate the AppleDouble files that result when macOS decides to add things like com.apple.quarantine to every JPG file that I download, lol)..

**Comment by arsaboo** [comment, marked as answer]
You may be able to use the [hook plugin](https://beets.readthedocs.io/en/stable/plugins/hook.html). It supports a range of events that you may be able to use.
